### ISSUE TITLE
zxpsave not restore xp in etlegacy x64 

### ISSUE THREAD

**Issue by oleksandr-l5** [Zelly/ZellyLuas#6]
after start new game XP not restoring
in console msg "unknown command userinfo"

**Comment by Zelly** [comment]
I do not have access to a legacy server any more(Nor was myn ever 64 bit) so I can't be much help there.
However "unknown command userinfo", that is in your client console correct? Can you look to see if the server is reporting any lua errors.

**Comment by oleksandr-l5** [comment]
I can, but dont know how to enable debug mode.

**Comment by oleksandr-l5** [comment]
I can attach zxpsave.log

**Comment by Zelly** [comment]
I was more talking about your etconsole.log since that should be where any lua errors would be popping up.
The more logs the merrier though, pastebin.com

**Comment by oleksandr-l5** [comment]
Dont know what happening, but now I have partial saving.
But when restoring or maybe saving have this in console:
"Lua API: et_ClientConnect error running lua script: [string "zxpsave.lua"]:334: bad argument #2 to 'G_XP_Set' (number has no integer representation)"
XP saving not all, for example I had 1005 XP in the end of mission, but after restoring I have 833.
Achieves(for example engeniring) not save or cant restore.

**Comment by Zelly** [comment]
Yeah that makes a little more sense.
Are you sure you are using latest version?
This was supposed to be fixed in latest version.

**Comment by oleksandr-l5** [comment]
Yes, but with some changes. I create pull request to show changes

**Comment by Zelly** [comment]
Yeah it really shouldn't be getting that error.
Honestly don't know what would be causing that error again.
Everything should be a float point at that point.
I guess in the xpsave.log when it shows up with loadxp (YOURGUIDHERE) 0.0 0.0... etc. are any of those values not float values, that will at least tell me if that part is wrong since I can't test it myself.
Lua always boggling you tell it specifically to check for certain things, and it still manages to sneak through somehow.
